### PR TITLE
Remove ArrayList From CorfuTable

### PR DESCRIPTION
**Pull request by Maithem**
Medhavi's tweaks to the CorfuTable consist of not using guava
for the internal data structures.

### CODE REVIEW

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **d378eb4**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #944 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/2UNqqRl419f0SLUociqFJZFj1tu5BytK)

File: runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java [hunk 1]
```diff
@@ -671,10 +671,10 @@ protected void unmapSecondaryIndexes(K key, V value) {
     @SuppressWarnings("unchecked")
     protected void mapSecondaryIndexes(K key, V value) {
         try {
-            Map.Entry<K, V> entry = new AbstractMap.SimpleImmutableEntry<>(key, value);
             indexFunctions.parallelStream()
                     .forEach(f -> f.getIndexFunction().generateIndex(key, value)
-                            .forEach(i -> indexMap.get(f).put((I) i, entry)));
+                            .forEach(i -> indexMap.get(f).computeIfAbsent((I) i,
+                                    (I) -> new HashMap<K, V>()).put(key, value)));
```
Comment (corfu-repo) [inline runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java]: MINOR Remove the parentheses around the "I" parameter [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)
MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)

File: runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java [hunk 2]
```diff
@@ -646,11 +647,10 @@ public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V>
     protected void unmapSecondaryIndexes(K key, V value) {
         try {
             if (value != null) {
-                final Map.Entry<K, V> previousEntry
-                        = new AbstractMap.SimpleImmutableEntry<>(key, value);
                 indexFunctions.parallelStream()
                         .forEach(f -> f.getIndexFunction().generateIndex(key, value)
-                                .forEach(i -> indexMap.get(f).remove((I) i, previousEntry)));
+                                .forEach(i -> indexMap.get(f).computeIfAbsent((I) i,
+                                        (I) -> new HashMap<K, V>()).remove(key)));
```
Comment (corfu-repo) [inline runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java]: MINOR Remove the parentheses around the "I" parameter [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)
MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)

**Comment by corfu-repo** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**